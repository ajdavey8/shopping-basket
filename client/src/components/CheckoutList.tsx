import React from 'react'

import CheckoutListItem from './CheckoutListItem';
import { List } from './styled-components'
import { Product } from '../types';


interface CheckoutList {
  products: Product[]
}

const CheckoutList = ({ products }: CheckoutList) => (
  <List>
    {products.map((product: Product) => (
      <CheckoutListItem
        key={product.id}
        {...product}
      />
    ))}
  </List>
)

export default CheckoutList