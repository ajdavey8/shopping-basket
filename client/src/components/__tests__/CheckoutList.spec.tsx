import React from 'react';
import { render } from '@testing-library/react'
import CheckoutList from '../CheckoutList'

const products = [
  {
    id: 1,
    name: 'soap',
    price: 2,
    image: 'pictureOfSoap',
    quantity: 1
  }
]

const setup = () => {
  const utils = render(
    <CheckoutList
      products={products}
    />
  )

  return {
    ...utils
  }
}

describe('<CheckoutList>', () => {
  it('should return product', () => {
    const { asFragment } = setup()
    expect(asFragment).toMatchSnapshot();
  })
})