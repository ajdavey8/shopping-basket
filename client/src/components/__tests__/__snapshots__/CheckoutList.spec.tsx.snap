// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CheckoutList> should return product 1`] = `[Function]`;
